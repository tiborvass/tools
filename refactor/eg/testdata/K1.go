// +build ignore

package K1

func f(x int, y ...interface{})                  {}
func g(a []interface{}, b int, c ...interface{}) {}

func example() {
	temp := 5
	args := []interface{}{3, false, "abc"}
	params := args

	f(temp)
	f(temp, 42)
	f(temp, 1, temp+temp, 2)
	f(temp, func() { f(7) })
	g(nil, 0, args)
	g(nil, 0, args...)
	f(temp, params)
	f(temp, params...)
}
