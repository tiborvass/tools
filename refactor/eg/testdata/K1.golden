// +build ignore

package K1

func f(x int, y ...interface{})                  {}
func g(a []interface{}, b int, c ...interface{}) {}

func example() {
	temp := 5
	args := []interface{}{3, false, "abc"}
	params := args

	g(nil, temp)
	g([]interface{}{42}, temp, 42)
	g([]interface{}{1, temp + temp, 2}, temp, 1, temp+temp, 2)
	g([]interface{}{func() { g(nil, 7) }}, temp, func() { g(nil, 7) })
	g(nil, 0, args)
	g(nil, 0, args...)
	g([]interface{}{params}, temp, params)
	g(params, temp, params...)
}
